homeassistant/helpers/event.py

```diff
@@ -381,17 +409,104 @@ def _process_state_match(parameter):
     return lambda state: state in parameter
 
 
-def _process_time_match(parameter):
-    """Wrap parameter in a tuple if it is not one and returns it."""
+def _parse_time_expression(parameter, min_value, max_value):
+    """Parse the time expression part and return a list of times to match."""
     if parameter is None or parameter == MATCH_ALL:
-        return lambda _: True
-
-    if isinstance(parameter, str) and parameter.startswith('/'):
+        res = [x for x in range(min_value, max_value + 1)]
+    elif isinstance(parameter, str) and parameter.startswith('/'):
         parameter = float(parameter[1:])
-        return lambda time: time % parameter == 0
+        res = [x for x in range(min_value, max_value + 1)
+               if x % parameter == 0]
```

Comment:
Not sure why this is a `float` (cron syntax doesn't allow it), but I added it here to prevent a potential breaking change.